### PR TITLE
Extract AddCommand Tests #327 

### PR DESCRIPTION
**Pull request by Zhiyuan-Amos**
Improves #327

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@Zhiyuan-Amos submitted v1 for review.

* [v1 1/4] [Refactor Integration Tests for AddCommand](https://github.com/se-edu/addressbook-level4/pull/387/commits/fdc2e0bad0ea87f64350e6a61e3de487fecda2b7)
* [v1 2/4] [Refactor helper methods into CommandTestUtil](https://github.com/se-edu/addressbook-level4/pull/387/commits/a3c503eec4137ba04384c7dac33da16c8df09a89)
* [v1 3/4] [Refactor Parser Tests for AddCommand](https://github.com/se-edu/addressbook-level4/pull/387/commits/cc300989aa19c33d060d3467af2676e4ff3f7bea)
* [v1 4/4] [Add Integration and Parser Tests for AddCommand](https://github.com/se-edu/addressbook-level4/pull/387/commits/8c933bd7c55efb8a9d23adaeb59241f4c3c9b80c)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/dc9fe96a38a2b5f9bb5bb4c05c3ff1db47b1effd...8c933bd7c55efb8a9d23adaeb59241f4c3c9b80c))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/387/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

File: src/test/java/seedu/address/testutil/ParserTestUtil.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ */
+public class ParserTestUtil {
+    private static Parser parser = new Parser();
+
+    public static void assertParseFailure(String userInput, String expectedMessage) {
+        Command command = parser.parseCommand(userInput);
+
+        // The current implementation of the parsing of commands will always return
+        // an IncorrectCommand if the parsing failed.
+        if (!(command instanceof IncorrectCommand)) {
```
Comment (MightyCupcakes) [inline src/test/java/seedu/address/testutil/ParserTestUtil.java]: Use assertions here instead?

Comment (Zhiyuan-Amos) [inline src/test/java/seedu/address/testutil/ParserTestUtil.java]: Yup true that, thanks!

**Comment by Zhiyuan-Amos** [comment]
@yamgent #317 as mentioned in this PR, should we put this PR on hold first? 

This PR contains the addition of a `ParserTestUtil` class which tests whether the parsing is successfully done. As such, merging this PR first will allow the testing of parsing for EditCommand to be done easily.

Alternatively, I can cherry pick that commit out into another branch and work on the testing of parsing for EditCommand.

**Comment by Zhiyuan-Amos** [comment]
@yamgent Following discussion with Huang Chao #327 , I will submit the 2nd iteration even though you haven't done the review yet, because the current architecture is different from the rest of the AddressBook.

**Comment by CanIHasReview** [comment]
# v2

@Zhiyuan-Amos submitted v2 for review.

* [v2 1/4] [Refactor Integration Tests for AddCommand](https://github.com/se-edu/addressbook-level4/pull/387/commits/fdc2e0bad0ea87f64350e6a61e3de487fecda2b7)
* [v2 2/4] [Refactor helper methods for Commands Tests](https://github.com/se-edu/addressbook-level4/pull/387/commits/ed5299c7acf9b0ac959444c9314d7510dfc7dcd7)
* [v2 3/4] [Refactor Parser Tests for AddCommand](https://github.com/se-edu/addressbook-level4/pull/387/commits/885d62b1ab8118945623d2c4a6b985b7aecb9841)
* [v2 4/4] [Add Integration and Parser Tests for AddCommand](https://github.com/se-edu/addressbook-level4/pull/387/commits/be77652cc02f7d9b774cb925330d73d3c6d25373)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/dc9fe96a38a2b5f9bb5bb4c05c3ff1db47b1effd...be77652cc02f7d9b774cb925330d73d3c6d25373)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/81a7e095ebb3c706c1dc645984c89681670cc013~1...81a7e095ebb3c706c1dc645984c89681670cc013))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/387/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by Zhiyuan-Amos** [comment]
Closing this PR because it seems to be too big.

**Pull request closed without merging**